### PR TITLE
Bugfix/server connection release

### PR DESCRIPTION
**Pull request by HerrMuellerluedenscheid**
Fixing https://github.com/Yakifo/amqtt/pull/101

### CODE REVIEW

**Comment by HerrMuellerluedenscheid** [comment]
I'm a little confused about the general exception handling in this block. I'm wondering if it wouldn't make more sense to change the base class to `AMQTTException` of  `NoDataException`, `MQTTException` and `CodecException` here https://github.com/Yakifo/amqtt/blob/master/amqtt/errors.py 
This would allow to catch them all at once. But I saw that the writer was not closed [here](https://github.com/Yakifo/amqtt/blob/master/amqtt/broker.py#L422) and I'm wondering what the reason was and if the writer needs to be closed in the added exception handling of `NoDataException` as well. But that can be tested/treated in another MR.

File: pyproject.toml [hunk 1]
```diff
@@ -26,6 +26,7 @@ passlib = "^1.7.0"
 docopt = "^0.6.0"
 PyYAML = ">=5.4.0,<7.0"
 coveralls = {version = "^3.0.1", optional = true}
+psutil = "^5.9.0"
```
Comment (HerrMuellerluedenscheid) [inline pyproject.toml:29]: this should be in dev dependencies

**Comment by not-f-elsner** [comment]
LGTM

**Review by not-f-elsner — APPROVED** [review]

File: amqtt/broker.py [hunk 2]
```diff
@@ -421,15 +421,24 @@ async def client_connected(
...
             await writer.close()
+            server.release_connection()
             self.logger.debug("Connection closed")
             return
+        except NoDataException as ne:
+            self.logger.error(
+                "No data from %s : %s"
+                % (format_client_message(address=remote_address, port=remote_port), ne)
+            )
+            server.release_connection()
```
Comment (not-f-elsner) [inline amqtt/broker.py:440]: Instead of calling `server.release_connection()` in every `except` block, maybe rather add a `finally` block for that

Comment (HerrMuellerluedenscheid) [reply amqtt/broker.py:440]: Thanks for the feedback. `server.release_connection()` is not called in the `try` case. Thus, moving the call to `finally` will not be the same behaviour.

Comment (not-f-elsner) [reply amqtt/broker.py:440]: Ah true, my bad :+1:

**Pull request merged**